frontend/packages/dev-console/src/components/import/pipeline/pipeline-template-utils.ts

```diff
@@ -0,0 +1,69 @@
+import * as _ from 'lodash';
+import { K8sResourceKind, k8sList, k8sGet, k8sCreate } from '@console/internal/module/k8s';
+import { PipelineModel, TaskModel } from '../../../models';
+import { GitImportFormData } from '../import-types';
+import { createPipelineResource } from '../../pipelines/pipeline-resource/pipelineResource-utils';
+
+export const getPipelineTaskRefs = (pipeline: K8sResourceKind): string[] => {
+  const taskRefs = _.get(pipeline, 'spec.tasks');
+  return taskRefs && taskRefs.map((task) => _.get(task, 'taskRef.name'));
+};
+
+export const getPipelineTemplate = async (runtime: string): Promise<K8sResourceKind> => {
+  const templates = await k8sList(PipelineModel, {
+    ns: 'openshift',
+    labelSelector: { 'pipeline.openshift.io/runtime': runtime },
+  });
+  return templates && templates[0];
+};
+
+export const copyPipelineTasks = (taskRefs: string[], namespace: string) => {
+  taskRefs.forEach(async (taskRef) => {
+    try {
+      const task = await k8sGet(TaskModel, taskRef, 'openshift');
+      task.metadata = { name: task.metadata.name, namespace };
+      await k8sCreate(TaskModel, task, namespace).catch(() => {});
+    } catch (err) {
+      throw err;
+    }
+  });
+};
+
+export const createGitResource = (url: string, ref: string = 'master', namespace: string) => {
```

Comment:
can we have params with default value as `ref` as the last argument and return type

Reply:
Done